client/state/analytics/reducer.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import { ANALYTICS_EVENT_RECORD } from 'state/action-types';
+
+/**
+ * Returns the event ids triggered after an action has been dispatched.
+ * This is usefull to avoid triggering those events twice
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const reducer = createReducer( {}, {
+	[ ANALYTICS_EVENT_RECORD ]: ( state, { meta: { analytics } } ) => {
+		return {
```

Comment:
There's a balance between performance and readability, but do you think storing the state as an array would be better or worse?

```js
return union( state, map( analytics, 'payload.eventId' ) );
```

Advantage being slightly easier to read and avoiding multiple instances of `Object.assign`, though the selector may be less performant (`Array#indexOf` being slightly slower than object lookup).